include/range/v3/view/group_by.hpp

```diff
@@ -53,56 +54,52 @@ namespace ranges
     private:
         friend range_access;
         Rng rng_;
+        // cached version of the end of the first subrange / start of the second subrange
+        detail::non_propagating_cache<iterator_t<Rng>> second_;
         semiregular_box_t<Fun> fun_;
 
-        template<bool IsConst>
+        struct pred
+        {
+            iterator_t<Rng> first_;
+            semiregular_box_ref_or_val_t<Fun, false> fun_;
+            bool operator()(range_reference_t<Rng> r) const
+            {
+                return invoke(fun_, *first_, r);
+            }
+        };
+
         struct cursor
         {
         private:
-            friend struct cursor<!IsConst>;
             friend range_access;
             friend group_by_view;
-            using CRng = meta::const_if_c<IsConst, Rng>;
-            iterator_t<CRng> cur_;
-            sentinel_t<CRng> last_;
-            semiregular_box_ref_or_val_t<Fun, IsConst> fun_;
+            iterator_t<Rng> cur_;
+            iterator_t<Rng> second_;
```

Comment:
This one really is `next_cur_`, while the view's one is `second_`. For a second, while looking at `next()`, I thought it was doing exactly the same as `views::cache1`.